### PR TITLE
Add email validation

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds email validation in editor.

### How to test the changes in this Pull Request:

1. Add or edit a newsletter 
3. Observe email is validated* in two email inputs:
    1. In sidebar "Newsletter Settings" section  in "From"->"Email" input
    1. in test email sending email input

\* it's not meant as a 100% email validation, just a convenience for the user, so they don't e.g. send a test email to an empty email

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by jeffersonrabb** [comment]
I'm getting an error on this branch that isn't occurring on `master`:

- create a new Newsletter
- send a test
- Observe this warning (I believe the warning is on `master`, just throwing it in here in case relevant):
```
react_devtools_backend.js:6 Warning: A component is changing a controlled input of type text to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
```
- Begin typing in the Sender Name field, observe this fatal:
```
index.js:322 Uncaught TypeError: Cannot read property 'length' of undefined
    at Sidebar.render (index.js:322)
    at finishClassComponent (react-dom.js?ver=16.9.0:17177)
    at updateClassComponent (react-dom.js?ver=16.9.0:17132)
    at beginWork$1 (react-dom.js?ver=16.9.0:18643)
    at HTMLUnknownElement.callCallback (react-dom.js?ver=16.9.0:341)
    at Object.invokeGuardedCallbackDev (react-dom.js?ver=16.9.0:391)
    at invokeGuardedCallback (react-dom.js?ver=16.9.0:448)
    at beginWork$$1 (react-dom.js?ver=16.9.0:23355)
    at performUnitOfWork (react-dom.js?ver=16.9.0:22346)
    at workLoopSync (react-dom.js?ver=16.9.0:22323)
```

**Comment by adekbadek** [comment]
@jeffersonrabb that should be fixed by 7ed1cc7406d6de1a5a322c77c330243c520735e9

File: src/editor/sidebar/index.js [hunk 1]
```diff
@@ -229,7 +229,7 @@ class Sidebar extends Component {
 					<Button
 						isLink
 						onClick={ () => this.updateSender( senderName, senderEmail ) }
-						disabled={ inFlight }
+						disabled={ inFlight || senderEmail.length ? ! hasValidEmail( senderEmail ) : false }
```
Comment (jeffersonrabb) [inline src/editor/sidebar/index.js:232]: Currently the button is enabled during "in flight." 

```suggestion
						disabled={
							inFlight ||
							! senderEmail.length ||
							! senderName.length ||
							! hasValidEmail( senderEmail )
						}
```

Comment (adekbadek) [reply src/editor/sidebar/index.js]: Fixed: a31550d789321e5d78752b776e27e4d370576937

**Comment by adekbadek** [comment]
> this change is made it then breaks comma delimited emails in the test field

and that's kind of a deal-breaker :D

**Comment by jeffersonrabb** [comment]
> and that's kind of a deal-breaker :D

We'd need to split around the comma and assess each one. Hardly an urgent need, we can address in a future PR if at all...

**Pull request merged**